### PR TITLE
Add client support for Swarm Logs and Swarm Config

### PR DESCRIPTION
**Pull request by ACoderLife**
This pull request finishes off work started by @The-TT-Hacker here:
https://github.com/dotnet/Docker.DotNet/pull/472

It could also close this:
https://github.com/dotnet/Docker.DotNet/pull/387

If you are happy with the API being client.Config and not client.Swarm.Config

and also surfaces the service logs.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/Docker.DotNet?pullRequest=589) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/Docker.DotNet?pullRequest=589) before we can accept your contribution.<br/><br/>:x: ACoderLife [sign now](https://cla.dotnetfoundation.org/dotnet/Docker.DotNet?pullRequest=589)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/Docker.DotNet?pullRequest=589) it.</sub>

**Comment by ACoderLife** [comment]
@galvesribeiro let me know I you need some unit tests against these changes.  Thanks.

**Comment by galvesribeiro** [comment]
Hello! Thanks for the contribution. 

> let me know I you need some unit tests against these changes. Thanks.

Can you please add some basic coverage to it?

Thank you!

**Review by HofmeisterAn — CHANGES_REQUESTED** [review]
Thank you for submitting this pull request. I have reviewed the code and overall it looks good. However, as @galvesribeiro mentioned, tests are missing.

File: src/Docker.DotNet/Endpoints/IConfigsOperations.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Docker.DotNet.Models;
+
+namespace Docker.DotNet
+{
+    public interface IConfigsOperations
```
[the remark this answers is not in the document]
Comment (ACoderLife) [reply src/Docker.DotNet/Endpoints/IConfigsOperations.cs:8]: Done

File: src/Docker.DotNet/Endpoints/SwarmOperations.cs [hunk 2]
```diff
@@ -1,227 +1,263 @@
...
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Net.Http;
+    using System.Text;
+    using System.Threading.Tasks;
+    using System.Threading;
+    using Models;
+    using System.IO;
+
+    internal class SwarmOperations : ISwarmOperations
```
[the remark this answers is not in the document]
Comment (ACoderLife) [reply src/Docker.DotNet/Endpoints/SwarmOperations.cs:13]: Do you mean lineendings?  I'm having trouble with this as it seems to me the repo has a mix of lineendings already?  I ran git ls-files --eol and see both CRLF and LF.
My attempts at normalizing have resulted in even more files changed.

File: src/Docker.DotNet/Endpoints/ConfigsOperations.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Docker.DotNet.Models;
+
+namespace Docker.DotNet
+{
+    internal class ConfigsOperations : IConfigsOperations
```
Comment (HofmeisterAn) [inline src/Docker.DotNet/Endpoints/ConfigsOperations.cs:10]: Don't we need tests for this class too?

Comment (ACoderLife) [reply src/Docker.DotNet/Endpoints/ConfigsOperations.cs]: Yes, I'll add them.

Comment (ACoderLife) [reply src/Docker.DotNet/Endpoints/ConfigsOperations.cs]: Added: Create, List, Inspect and Remove tests.

**Review by HofmeisterAn — APPROVED** [review]
Looks good @galvesribeiro WDYT?

**Comment by ACoderLife** [comment]
@galvesribeiro Did you want me to make any other clean up?

**Comment by galvesribeiro** [comment]
Thanks for the PR! I would rename swarm-related objects as suggested and also fix the test. Everything else LGTM.

**Comment by ACoderLife** [comment]
@galvesribeiro renaming based on suggestions should be complete.  Fixed test.  Cheers.

**Comment by galvesribeiro** [comment]
All good! Thank you @ACoderLife. Will cut a new build soon.

**Pull request merged**